Make must-gather not collect sensitive resources by default (#3046)
scylladb/scylla-operator: docs/source/support/must-gather.md

# Gathering data with must-gather

`must-gather` is an embedded tool in Scylla Operator that helps collecting all the necessary info when something goes wrong.
The tool talks to the Kubernetes API, retrieves a predefined set of resources and saves them into a folder in your current directory.

:::{note}

> By default, sensitive resources like `Secrets` or `bitnami.com.SealedSecrets` are omitted from the collection.

Reviewer: On a second thought - is this note box necessary given that this page is not that long and is likely to be read in one go?

> If you ever need to include them (ScyllaDB support will explicitly ask for that if a need arises), please refer to the [include sensitive resources](#including-sensitive-resources) section below.

Reviewer: The reference to ScyllaDB Support is problematic because operator per se is OSS. Suggested rewording.

```suggestion
If you come across a clear need to include them (not recommended unless the default settings are deemed insufficient for investigation), please refer to the [include sensitive resources](#including-sensitive-resources) section below.
```

If you have other resources that you wouldn't like to include in the collection, refer to the [excluding resources](#excluding-resources) section below.
:::

> Given `must-gather` needs to talk to the Kubernetes API, at the very least, you need to supply the `--kubeconfig` flag with a path to the kubeconfig file for your Kubernetes cluster, or set the `KUBECONFIG` environment variable.

Reviewer: ```suggestion
Given that `must-gather` needs to talk to the Kubernetes API, at the very least, you need to supply the `--kubeconfig` flag with a path to the kubeconfig file for your Kubernetes cluster, or set the `KUBECONFIG` environment variable.
```

```bash
scylla-operator must-gather --all-resources
```

### Including sensitive resources

By default, `must-gather` omits sensitive resources like `Secrets` or `bitnami.com.SealedSecrets`. You can turn this
behavior off by passing the `--include-sensitive-resources` flag.

```bash
scylla-operator must-gather --all-resources --include-sensitive-resources
```

### Excluding resources

If you have resources that you don't want to include in the collection other than the default ones, you can exclude them
by passing the `--exclude-resource` flag multiple times.

:::{note}

> Format for the resource is `kind[.group]`, e.g., `LimitRange` or `DeviceClass.resource.k8s.io`. Group should be omitted for core resources.

Reviewer: More readable this way?

```suggestion
The format for the resource is `kind` (for core resources) or `kind.group`. Examples: `LimitRange` or `DeviceClass.resource.k8s.io`.
```
